spec/unit/lib/cloud_controller/permissions_spec.rb

```diff
@@ -661,121 +666,103 @@ module VCAP::CloudController
     end
 
     describe '#can_write_to_active_space?' do
+      let(:membership) { instance_double(Membership) }
+
+      before do
+        allow(Membership).to receive(:new).with(user).and_return(membership)
+      end
       context 'user has no membership' do
+        before do
+          allow(membership).to receive(:has_any_roles?).with(Permissions::ROLES_FOR_SPACE_WRITING, space.id).and_return(false)
+        end
         context 'and user is an admin' do
           it 'returns true' do
             set_current_user(user, { admin: true })
-            expect(permissions.can_write_to_active_space?(space_guid)).to be true
+            expect(permissions.can_write_to_active_space?(space.id)).to be true
           end
         end
 
         context 'and user is admin_read_only' do
           it 'return false' do
             set_current_user_as_admin_read_only
-            expect(permissions.can_write_to_active_space?(space_guid)).to be false
+            expect(permissions.can_write_to_active_space?(space.id)).to be false
           end
         end
 
         context 'and user is global auditor' do
           it 'return false' do
             set_current_user_as_global_auditor
-            expect(permissions.can_write_to_active_space?(space_guid)).to be false
+            expect(permissions.can_write_to_active_space?(space.id)).to be false
           end
         end
 
         context 'and user is not an admin' do
           it 'return false' do
             set_current_user(user)
-            expect(permissions.can_write_to_active_space?(space_guid)).to be false
+            expect(permissions.can_write_to_active_space?(space.id)).to be false
           end
         end
       end
 
       context 'user has valid membership' do
-        it 'returns true for space developer' do
```

Comment:
The previous checks for different roles also worked fine. Any reason why not to keep them?

Reply:
See my other comment [here](https://github.com/cloudfoundry/cloud_controller_ng/pull/3021/files#r1004261111)

Reply:
(but I'm switching this back in the way you suggest)